x-pack/legacy/plugins/apm/cypress/plugins/index.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// ***********************************************************
+// This example plugins/index.js can be used to load plugins
+//
+// You can change the location of this file or turn off loading
+// the plugins file with the 'pluginsFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/plugins-guide
+// ***********************************************************
+
+// This function is called when a project is opened or re-opened (e.g. due to
+// the project's config changing)
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+const wp = require('@cypress/webpack-preprocessor');
+const fs = require('fs');
+
+module.exports = on => {
+  // add typescript support
+  const options = {
+    webpackOptions: {
+      resolve: {
+        extensions: ['.ts', '.tsx', '.js']
+      },
+      module: {
+        rules: [
+          {
+            test: /\.tsx?$/,
+            loader: 'ts-loader',
+            options: { transpileOnly: true }
+          }
+        ]
+      }
+    }
+  };
+  on('file:preprocessor', wp(options));
+
+  // readFileMaybe
+  on('task', {
+    readFileMaybe(filename) {
```

Comment:
what's this for?

Reply:
Relative snapshots files required by the snapshot module: https://github.com/cypress-io/snapshot#userelativesnapshots